app/src/main/java/com/realityexpander/tasky/presentation/login_screen/LoginScreen.kt

```diff
@@ -57,113 +86,201 @@ fun LoginScreen(
-            }
-        )
-        if(loginState.isInvalidEmail) {
-            Text(text = UiText.Res(R.string.error_invalid_email).get(), color = Color.Red)
-        }
-        Spacer(modifier = Modifier.height(8.dp))
-
-        // PASSWORD
-        PasswordField(
-            value = loginState.password,
-            isError = loginState.isInvalidPassword,
-            onValueChange = {
-                viewModel.sendEvent(LoginEvent.UpdatePassword(it))
-            },
-            isPasswordVisible = loginState.isPasswordVisible,
-            clickTogglePasswordVisibility = {
-                viewModel.sendEvent(
-                    LoginEvent.TogglePasswordVisibility(loginState.isPasswordVisible)
-                )
-            },
-            imeAction = ImeAction.Done,
-            doneAction = {
-                performLogin()
-            }
+            .background(color = MaterialTheme.colors.onSurface)
+    ) col1@ {
+        Spacer(modifier = Modifier.largeHeight())
+        Text(
+            text = UiText.Res(R.string.login_title).get,
+            style = MaterialTheme.typography.h5,
+            fontWeight = FontWeight.Bold,
+            color = MaterialTheme.colors.surface,
+            modifier = Modifier
+                .align(alignment = Alignment.CenterHorizontally)
         )
-        if(loginState.isInvalidPassword) {
-            Text(text = UiText.Res(R.string.error_invalid_password).get(), color = Color.Red)
-            Spacer(modifier = Modifier.height(8.dp))
-        }
+        Spacer(modifier = Modifier.mediumHeight())
 
-        // LOGIN BUTTON
-        Spacer(modifier = Modifier.height(16.dp))
-        Button(onClick = {
-                performLogin()
-            },
-            enabled = !loginState.isLoading,
+        Column(
             modifier = Modifier
-                .align(alignment = Alignment.End)
+                .verticalScroll(rememberScrollState())
+                .taskyScreenTopCorners(color = MaterialTheme.colors.surface)
+                .weight(1f)
         ) {
-            Text(text = UiText.Res(R.string.login_button).get())
-            if(loginState.isLoading) {
-                CircularProgressIndicator(
-                    modifier = Modifier
-                        .padding(start = 4.dp)
-                        .size(16.dp)
-                        .align(alignment = CenterVertically)
+            Spacer(modifier = Modifier.mediumHeight())
+
+            // • EMAIL
+            EmailField(
+                value = loginState.email,
+                label = null,
+                isError = loginState.isInvalidEmail,
+                onValueChange = {
+                    viewModel.sendEvent(LoginEvent.UpdateEmail(it))
+                }
+            )
+            if (loginState.isInvalidEmail && loginState.isShowInvalidEmailMessage) {
+                Text(text = UiText.Res(R.string.error_invalid_email).get, color = Color.Red)
```

Comment:
If you construct the string res directly in the UI, you can also just use `stringResource(R.string.error_invalid_email)`

Reply:
Wasnt sure about this. 

OK I will eliminate the use of UiText for the composable.